libraries/AC_Avoidance/AP_OABendyRuler.cpp

```diff
@@ -144,11 +192,57 @@ bool AP_OABendyRuler::update(const Location& current_loc, const Location& destin
     destination_new.offset_bearing(chosen_bearing, distance_to_dest);
 
     // log results
-    AP::logger().Write_OABendyRuler(true, chosen_bearing, best_margin, destination, destination_new);
+    AP::logger().Write_OABendyRuler(true, chosen_bearing, false, best_margin, destination, destination_new);
 
     return true;
 }
 
+/*
+This function is called when BendyRuler has found a bearing which is obstacles free at atleast lookahead_step1_dist and  then lookahead_step2_dist from the present location
+In many situations, this new bearing can be either left or right of the obstacle, and BendyRuler can have a tough time deciding between the two.
+It has the tendency to move the vehicle back and forth, if the margin obtained is even slightly better in the newer iteration.
+Therefore, this method attempts to avoid changing direction of the vehicle by more than _bendy_angle degrees, 
+unless the new margin is atleast _bendy_ratio times better than the margin with previously calculated bearing.
+We return true if we have resisted the change and will follow the last calculated bearing. 
+*/
+bool AP_OABendyRuler::avoid_bearing_change(const Location &destination, const Location &current_loc, const bool &active, const float &bearing_test, const float &lookahead_step1_dist, const float &margin, float &final_bearing, float &final_margin) 
```

Comment:
let's remove all the "const" from the variables.. sorry, it just makes the argument list too long.

Reply:
Can't really do much here since I am passing these values by reference and things like destination, current_loc were passed to BendyRuler as a const. I removed const from where ever I could.
If you want I can pass them as a direct variable instead?

Reply:
We shouldn't pass standard types as const reference normally, specially for boolean that means passing the pointer address that is bigger that the value...

Reply:
Alright, fixed! Thanks